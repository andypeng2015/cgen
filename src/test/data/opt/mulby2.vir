module foo

export function w $foo(w %x) {
@start:
  %x = mul.w %x, 2_w
  ret %x
}
