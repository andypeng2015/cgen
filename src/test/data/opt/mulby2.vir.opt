module foo

export function w $foo(w %x) {
@2:
  %1 = lsl.w %x, 0x1_w ; @5
  ret %1
}
