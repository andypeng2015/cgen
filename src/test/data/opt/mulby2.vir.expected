module foo

export function w $foo(w %x) {
@2:
  %0 = lsl.w %x, 0x1_w ; @4
  ret %0
}
